crm_compassion/models/event_compassion.py

```diff
@@ -350,9 +340,19 @@ def onchange_type(self):
 
     @api.onchange('start_date')
     def onchange_start_date(self):
+        """ Update end_date and hold_start_date as soon as start_date is
+        changed """
         for event in self.filtered(lambda e: e.start_date and not e.end_date):
             event.end_date = event.start_date
 
+        for event in self.filtered('start_date'):
+            days_allocate_before_event = event.env[

```

Comment:
You can declare this variable outside of the loop to avoid recomputing it at each iteration.